### ISSUE TITLE
Hot reloading on file changes in docker no longer working in typescript 4.9.5

### ISSUE THREAD

**Issue by andersd1** [microsoft/TypeScript#52712]
# Bug Report
After upgrading to typescript 4.9.5, hot reloading stopped working in a docker environment. In our use case, we are using nestjs with command `nest start xx --watch --tsc`. I saw in the issue linked below that there have been changes to the file system. Is there anything docker/nest/typescript can do to get it working again ?

https://github.com/docker/compose/issues/10027

### 🔎 Search Terms
docker

### 🕗 Version & Regression Information
After upgrading to typescript 4.9.5

### 🙁 Actual behavior

Docker no longer detecting file changes

### 🙂 Expected behavior

Docker should detect file changes

**Comment by jakebailey** [comment]
After upgrading TypeScript to 4.9.5 from which version?

**Comment by andersd1** [comment]
From 4.8.4

**Comment by submarines-and** [comment]
Maybe this is what changed?

https://github.com/microsoft/TypeScript/pull/50366

**Comment by RyanCavanaugh** [comment]
Yeah, we changed the default filesystem watch implementation to one that's generally more efficient. It's possible your OS/FS/etc environment doesn't support this, however. You can set the `"watchFile"` setting in your tsconfig to something else; the default is new `"usefsevents"`, but `"fixedpollinginterval"` is likely to work anywhere.

**Comment by andersd1** [comment]
fixedpollinginterval worked fine. 
Thanks!

**Comment by matthandlersux** [comment]
posting a recap and adding a few extra keywords since it took me a bit to track down this issue from google... for anyone else that ends up here due to an error that looks like:
```
error TS6053: File '/path/to/real/file.ts' not found.
```
i believe the new way that file watching works does not work on macos catalina (10.15.x, and possibly below), so the suggestion above was required. to be more explicit: add to your `tsconfig.json`:
```json
  "compilerOptions": {
    ...
  },
  "watchOptions": {
    "watchFile": "fixedpollinginterval",
  },
```
alternatively you can set the following environment variable:
```bash
export TSC_WATCHFILE="PriorityPollingInterval"
```
and [here's the section about it in the docs](https://www.typescriptlang.org/docs/handbook/configuring-watch.html)

**Comment by nguyenthanhan201** [comment]
> Yeah, we changed the default filesystem watch implementation to one that's generally more efficient. It's possible your OS/FS/etc environment doesn't support this, however. You can set the `"watchFile"` setting in your tsconfig to something else; the default is new `"usefsevents"`, but `"fixedpollinginterval"` is likely to work anywhere.

Thank you. You saved my day.
